### PR TITLE
download ReducedDatum not ingested with files

### PR DESCRIPTION
**Pull request by griffin-h**
This takes advantage of the data sharing form to provide a method of downloading photometry to a text file. Right now there's no way to download data not ingested via a file, as noted in #248. This may not be the preferred way to implement this feature request, but I'm opening a PR in case any part of the code I wrote is useful, either as an interim solution or to be reused in a better solution. Also feel free to close without merging!

To use this feature, select data from the sharing form (or select all), choose "download" as the destination, and click "submit." The "title" field will become the filename, and the "message" field will become a comment at the top of the file. The output table contains all the fields in the `ReducedDatum` table, not all of which are useful, including the superset of all keys in the `value` `JSONField`.

### CODE REVIEW

File: tom_dataproducts/views.py [hunk 1]
```diff
@@ -30,7 +30,7 @@
 from tom_dataproducts.data_processor import run_data_processor
 from tom_observations.models import ObservationRecord
 from tom_observations.facility import get_service_class
-from tom_dataproducts.sharing import share_data_with_hermes, share_data_with_tom, sharing_feedback_handler
+from tom_dataproducts.sharing import share_data_with_hermes, share_data_with_tom, sharing_feedback_handler, download_data
```
Comment (jchate6) [inline tom_dataproducts/views.py:33]: Can you split this line to make Lint Happy?

File: tom_dataproducts/sharing.py [hunk 2]
```diff
@@ -293,3 +298,21 @@ def sharing_feedback_handler(response, request):
     else:
         messages.success(request, publish_feedback)
     return
+
+
+def download_data(form_data, selected_data):
```
Comment (jchate6) [inline tom_dataproducts/sharing.py:303]: Can we get a short docstring describing this function?

**Review by jchate6 — CHANGES_REQUESTED** [review]
This is great. Thanks for this.
Just a few very minor changes

File: tom_dataproducts/views.py (hunk 1 above)
Comment (griffin-h) [reply tom_dataproducts/views.py]: Done!

File: tom_dataproducts/sharing.py (hunk 2 above)
Comment (griffin-h) [reply tom_dataproducts/sharing.py:303]: Done! Let me know if it's clear.

Comment (jchate6) [reply tom_dataproducts/sharing.py:303]: excellent. Thanks.

**Review by jchate6 — APPROVED** [review]
very nice.

**Pull request merged**